static/js/spoilers.js

```diff
@@ -0,0 +1,67 @@
+function collapse_spoiler(spoiler) {
+    const spoiler_height = spoiler.prop('scrollHeight');
+
+    // Set height to rendered height on next frame, then to zero on following
+    // frame to allow CSS transition animation to work
+    requestAnimationFrame(function () {
+        spoiler.height(spoiler_height + 'px');
+        spoiler.removeClass("spoiler-content-open");
+
+        requestAnimationFrame(function () {
+            spoiler.height("0px");
+        });
+    });
+}
+
+function expand_spoiler(spoiler) {
+    // Normally, the height of the spoiler block is not defined absolutely on
+    // the `spoiler-content-open` class, but just set to `auto` (i.e. the height
+    // of the content). CSS animations do not work with properties set to
+    // `auto`, so we get the actual height of the content here and temporarily
+    // put it explicitly on the element styling to allow the transition to work.
+    const spoiler_height = spoiler.prop('scrollHeight');
+    spoiler.height(spoiler_height + "px");
+    // The `spoiler-content-open` class has CSS animations defined on it which
+    // will trigger on the frame after this class change.
+    spoiler.addClass("spoiler-content-open");
+
+    spoiler.on('transitionend', function () {
+        spoiler.off('transitionend');
+        // When the CSS transition is over, reset the height to auto
+        // This keeps things working if, e.g., the viewport is resized
+        spoiler.height("");
+    });
+}
+
+exports.initialize = function () {
+    $("#home").on("click", ".spoiler-button", function () {
```

Comment:
This should be on `body`, not home; otherwise, it doesn't work in the markdown help.  Also needs `stopPropagation` / `preventDefault`.

(I fixed both)